com.hazelcast.client.listeners.MultiMapEntryListenerOnReconnectTest.testListenersTerminateRandomNode_smart [API-1656]

_5.1.z_ (commit 6caccfd1b9c4bcbe50bbbcca554594e83864b8c7)

Failed on Openj9-11: https://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-5.maintenance-Openj9-11/53/testReport/com.hazelcast.client.listeners/MultiMapEntryListenerOnReconnectTest/testListenersTerminateRandomNode_smart/

<details><summary>Stacktrace:</summary>

```
java.lang.AssertionError: Received event count is 9 but it is expected to stay at 10 expected:<10> but was:<9>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest.lambda$validateListenerFunctionality$4(AbstractListenersOnReconnectTest.java:399)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest$$Lambda$6259/0x0000000000000000.run(Unknown Source)
	at com.hazelcast.test.HazelcastTestSupport.assertTrueAllTheTime(HazelcastTestSupport.java:1181)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest.validateListenerFunctionality(AbstractListenersOnReconnectTest.java:397)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest.validateRegistrationsAndListenerFunctionality(AbstractListenersOnReconnectTest.java:332)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest.testListenersTerminateRandomNode(AbstractListenersOnReconnectTest.java:139)
	at com.hazelcast.client.listeners.AbstractListenersOnReconnectTest.testListenersTerminateRandomNode_smart(AbstractListenersOnReconnectTest.java:105)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:866)
```

</details>

Standard output can be found here - https://console.aws.amazon.com/s3/buckets/j-artifacts/Hazelcast-5.maintenance-Openj9-11/53/

--- a/hazelcast/src/test/java/com/hazelcast/client/listeners/AbstractListenersOnReconnectTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/client/listeners/AbstractListenersOnReconnectTest.java
@@ -385,7 +385,7 @@ public abstract class AbstractListenersOnReconnectTest extends ClientTestSupport
         eventCount.set(0);
         eventsLatch = new CountDownLatch(1);
         for (int i = 0; i < EVENT_COUNT; i++) {
-            events.add(randomString());
+            events.add(i + randomString());
         }
 
         for (String event : events) {
@@ -430,8 +430,8 @@ public abstract class AbstractListenersOnReconnectTest extends ClientTestSupport
     protected abstract void produceEvent(String event);
 
     void onEvent(String event) {
-        events.remove(event);
         eventCount.incrementAndGet();
+        events.remove(event);
         if (events.isEmpty()) {
             eventsLatch.countDown();
         }

